lib/utils.js

```diff
@@ -46,29 +46,45 @@ var parseUrl = function (href) {
 };
 
 var redefine = function (obj, prop, value) {
+
     if (obj[prop]) {
         // TODO, consider erasing scope/hiding (enumerable: false)
         obj["_" + prop] = obj[prop];
     }
+
     Object.defineProperty(obj, prop, {
-        enumerable: obj.propertyIsEnumerable(prop),
         value: value,
+        enumerable: obj.propertyIsEnumerable(prop),
         configurable: true
     });
 };
 
+var defineGetter = function (obj, prop, getter) {
+
+    if (obj[prop]) {
+        // TODO, consider erasing scope/hiding (enumerable: false)
+        obj["_" + prop] = obj[prop];
+    }
+
+    Object.defineProperty(obj, prop, {
```

Comment:
Opportunity to cache the property descriptor and reuse it by just setting .get = getter on it before passing to defineProperty()

Reply:
Sorry I don't understand about cache?

Reply:
This is not needed for now.

Reply:
I store previous value because it's not a define not because of saving previous define.
I need in that case reponse before it was overide in case I reuse XHR.